tank/albert_lite_base/albert_lite_base_tflite_test.py

```diff
@@ -70,58 +70,26 @@ def __init__(
     def create_and_check_module(self):
         shark_args.save_mlir = self.save_mlir
         shark_args.save_vmfb = self.save_vmfb
-
-        # Preprocess to get SharkImporter input args
-        tflite_preprocessor = TFLitePreprocessor(model_name="albert_lite_base")
-        raw_model_file_path = tflite_preprocessor.get_raw_model_file()
-        inputs = tflite_preprocessor.get_inputs()
-        tflite_interpreter = tflite_preprocessor.get_interpreter()
-
-        # Use SharkImporter to get SharkInference input args
-        my_shark_importer = SharkImporter(
-            module=tflite_interpreter,
-            inputs=inputs,
-            frontend="tflite",
-            raw_model_file=raw_model_file_path,
-        )
-        mlir_model, func_name = my_shark_importer.import_mlir()
-
-        # Use SharkInference to get inference result
-        shark_module = SharkInference(
-            mlir_module=mlir_model,
-            function_name=func_name,
-            device=self.device,
-            mlir_dialect="tflite",
+        shark_downloader = SharkDownloader(
```

Comment:
> and instead of creating a class for the downloader, we could create a simple function like:
> 
> https://github.com/nod-ai/SHARK/blob/f3864c2639e6ce3b4eb4163083e14d68e331ed43/shark/shark_downloader.py#L31

No. If the model has been downloader before, not need to download it again. I think Anush wants a general API. And for tflite, since all input is json, it will have to give the input type,  not just model name. Or I will have to change tons of JSON processes.

Reply:
> Can we make it minimal and just assume that all the models are present in shark_tank? I have just added the downloader for torch here: https://github.com/nod-ai/SHARK/blob/f3864c2639e6ce3b4eb4163083e14d68e331ed43/shark/examples/shark_inference/minilm_jit.py This is what we need to expose to the users and they just would just want to pass the model name.

No. If the model has been downloader before, not need to download it again. I think Anush wants a general API. And for tflite, since all input is json, it will have to give the input type, not just model name. Or I will have to change tons of JSON processes.